### PR TITLE
Make catcher great again!

### PR DESCRIPTION
**Pull request by gavinlaking**
- Bump Ruby version to 2.3
- Remove EventMachine
- Fix string encoding

### CODE REVIEW

File: spec/catcher_spec.rb [hunk 1]
```diff
@@ -16,21 +16,28 @@
     let(:options) { { :id => id, :locale => locale } }
     let(:cache_key) { 'example-en-1' }
     let(:resource) { "http://example.com/en/1" }
-    let(:request) { stub(response:'{"example":{"id":1}}') }
+    let(:request) { stub(response: response, body: response) }
```
Comment (michaelpreston) [inline spec/catcher_spec.rb]: Prefer `double` over `stub` as we're using a newer version of RSpec

File: spec/catcher_spec.rb [hunk 2]
```diff
@@ -16,21 +16,28 @@
...
+    let(:response) { '{"example":{"id":1}}' }
+
+    let(:http) { stub }
+
+    before do
+      ::Net::HTTP.stubs(:new).returns(http)
```
Comment (michaelpreston) [inline spec/catcher_spec.rb]: `allow(Net::HTTP).to receive(:new).and_return(http)` etc

File: spec/catcher_spec.rb (hunk 1 above)
Comment (gavinlaking) [inline spec/catcher_spec.rb]: If we use `double` then we get `undefined method `double' for #<RSpec::ExampleGroups::Catcher...>`, which can be remedied by removing the line `config.mock_framework = :mocha` from the `spec_helper.rb` file. The downside of this approach is that we then have to fix all the tests for Catcher making this a much larger pull request.

I'd rather keep my footprint as low as possible for now, and provide a subsequent pull request in the future to upgrade all of the tests to our preferred syntax.

File: spec/catcher_spec.rb (hunk 2 above)
Comment (gavinlaking) [inline spec/catcher_spec.rb]: As above.

**Pull request merged**